js/private/js_binary.sh.tpl

```diff
@@ -173,15 +172,24 @@ function normalize_windows_path {
 if [ "${TEST_SRCDIR:-}" ]; then
     # Case 4, bazel has identified runfiles for us.
     RUNFILES="$TEST_SRCDIR"
-elif [ "${RUNFILES_MANIFEST_ONLY:-}" ]; then
-    # Windows only has a manifest file instead of symlinks.
+elif [ "${RUNFILES_MANIFEST_FILE:-}" ]; then
     if [ "$(is_windows)" -eq "1" ]; then
-        # If Windows normalizing the path and case insensitive removing the `/MANIFEST` part of the path
-        NORMALIZED_RUNFILES_MANIFEST_FILE_PATH=$(normalize_windows_path "$RUNFILES_MANIFEST_FILE")
-        # shellcheck disable=SC2001
-        RUNFILES=$(sed 's|\/MANIFEST$||i' <<< "$NORMALIZED_RUNFILES_MANIFEST_FILE_PATH")
+        # If Windows, normalize the path
+        NORMALIZED_RUNFILES_MANIFEST_FILE=$(normalize_windows_path "$RUNFILES_MANIFEST_FILE")
     else
-        RUNFILES=${RUNFILES_MANIFEST_FILE%/MANIFEST}
+        NORMALIZED_RUNFILES_MANIFEST_FILE="$RUNFILES_MANIFEST_FILE"
+    fi
+    if [[ "${NORMALIZED_RUNFILES_MANIFEST_FILE}" == *.runfiles_manifest ]]; then
+        # Newer versions of Bazel put the manifest besides the runfiles with the suffix .runfiles_manifest.
+        # For example, the runfiles directory is named my_binary.runfiles then the manifest is beside the
+        # runfiles directory and named my_binary.runfiles_manifest
+        RUNFILES=${NORMALIZED_RUNFILES_MANIFEST_FILE%_manifest}
+    elif [[ "${NORMALIZED_RUNFILES_MANIFEST_FILE}" == */MANIFEST ]]; then
+        # Older versions of Bazel put the manifest file named MANIFEST in the runfiles directory
```

Comment:
Do you know what version switched it? Can we put a comment so we can potentially remove the `elif` in the future?

Reply:
Not sure and doesn't seem worthwhile to do a binary search for this one